packages/core/helper-plugin/src/components/AnErrorOccurred.tsx

```diff
@@ -2,10 +2,24 @@ import React from 'react';
 
 import { EmptyStateLayout } from '@strapi/design-system';
 import { ExclamationMarkCircle } from '@strapi/icons';
-import PropTypes from 'prop-types';
-import { useIntl } from 'react-intl';
+import { PrimitiveType, useIntl } from 'react-intl';
 
-const AnErrorOccurred = ({ content, ...rest }) => {
+export type AnErrorOccurredProps = {
+  content?: {
+    id: string;
+    defaultMessage: string;
+    values?: Record<string, PrimitiveType>;
+  };
+};
+
+const AnErrorOccurred: React.FC<AnErrorOccurredProps> = ({
```

Comment:
> think it's better don't use React.FC, it accepts children even if in this case we should not allow childrens.

this was react17 only, but yes I think the paradigm we're going for is not using `React.FC` at all, we don't in the design-system :)

Reply:
> Also, wdyt about extends `EmptyStateLayoutProps` from Design-System?

I don't feel safe about this. Should I extend the props with something like  `Omit< EmptyStateLayoutProps, 'content' | 'icon' >;` to match?

Reply:
I also think about to have a global type in the helper-plugin for `MessageDescriptor & { values?: Record<string, PrimitiveType> }`. I need to use this in some other files too, for example in `NoContent.jsx` which has similar props.

Where would you put those definitions?

Reply:
I would do the `Omit<EmptyStateLayoutProps, 'content' | 'icon'>` and then swap the order of the props like:

```
<EmptyStateLayout
      {...rest}
      icon={<ExclamationMarkCircle width="10rem" />}
      content={formatMessage(
        { id: content.id, defaultMessage: content.defaultMessage },
        content.values
      )}
    />
```

So people can't reset them, and we consider this to be a bug if it were used this way :)

@nolliebigspin to answer your question on:

```ts
type X = MessageDescriptor & { values?: Record<string, PrimitiveType> }
```

I think if it's used multiple places then yes, lets just share it because it's more straight forward. I'll leave it to you all to come up with a good name ⭐

Reply:
Okay, I pushed the extending prop changes.

```ts
type X = MessageDescriptor & { values?: Record<string, PrimitiveType> }
```

For this type I'll wait until all other components using this are done and refer it in another separate PR